lib/mastodon/cli/maintenance.rb

```diff
@@ -546,7 +546,7 @@ def deduplicate_tags!
       if ActiveRecord::Migrator.current_version < 2021_04_21_121431
         ActiveRecord::Base.connection.add_index :tags, 'lower((name)::text)', name: 'index_tags_on_name_lower', unique: true
       else
-        ActiveRecord::Base.connection.execute 'CREATE UNIQUE INDEX CONCURRENTLY index_tags_on_name_lower_btree ON tags (lower(name) text_pattern_ops)'
```

Comment:
This is the one remaining thing in running this command which cannot exist within a transaction (the index portion is fine, its the `concurrently` specifically. Ideas for resolving:

- If this actually doesn't matter for purposes of this script ... then my change here to just remove it is fine maybe?
- If this does matter and needs to stay, the next best thing I can think of is to do a very narrowly targetted stub here where we intercept just this one index creation call (might need to move to isolated method) and have it just not do this one during the spec run
- Along those same lines, but do some monkey-patching of either `execute` or `add_index` or something where just in this one spec we silently drop the `concurrently`, but in actual usage it would run.

Reply:
Ah, I think we can drop the `CONCURRENTLY`, as you're not supposed to be running Mastodon and the maintenance script at the same time anyway!

Reply:
Excellent - that will allow this all to be transaction-wrapped with this approach. I'll leave the change in place.